modules/data-entry/src/main/frontend/src/questionnaire/Section.jsx

```diff
@@ -0,0 +1,82 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Grid, Typography, withStyles } from "@material-ui/core";
+import uuidv4 from "uuid/v4";
+
+import FormEntry, { ENTRY_TYPES } from "./FormEntry";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+/**
+ * Component that consists of a few sections, and optionally has some criteria to its display
+ *
+ * @example
+ * <Section depth={1} path="." sectionDefinition={{label: "Section"}} />
+ *
+ * @param {int} depth the section nesting depth
+ * @param {Object} existingAnswer form data that may include answers already submitted for this component
+ * @param {string} path the path to the parent of the question
+ * @param {Object} sectionDefinition the section definition JSON
+ */
+function Section(props) {
+  const { classes, depth, existingAnswer, path, sectionDefinition } = props;
+  const [sectionID] = useState((existingAnswer && existingAnswer[0]) || uuidv4());
+  const sectionPath = path + "/" + sectionID;
+  const headerVariant = (depth > 3 ? "body1" : ("h" + (depth+3)));
```

Comment:
The reasoning behind this could be made easier to understand, something along the lines of:
```suggestion
  const headerVariant = (depth > (MAX_HEADING_LEVEL - MIN_HEADING_LEVEL) ? "body1" : ("h" + (depth+MIN_HEADING_LEVEL)));
```
where MAX_HEADING_LEVEL is 6 and MIN_HEADING_LEVEL comes from the form and is 1 + The heading level of the title.

Or at least add a comment :)